bmorph/util/mizuroute_utils.py

```diff
@@ -135,20 +135,83 @@ def map_ref_sites(routed: xr.Dataset, gauge_reference: xr.Dataset,
             cur_seg = routed['up_seg'].sel(seg=cur_seg).values[()]
         if cur_seg in routed['seg'].values:
             routed['up_ref_seg'].loc[{'seg':seg}] = routed['up_ref_seg'].sel(seg=cur_seg).values[()]
-
+            
     # Fill in any remaining nulls (head/tailwaters)
-    if fill_method == 'forward_fill':
-        routed['up_ref_seg'] = (routed['up_ref_seg'].where(
-            ~np.isnan(routed['up_ref_seg']), other=routed['down_ref_seg'])).ffill('seg')
-        routed['down_ref_seg'] = (routed['down_ref_seg'].where(
-            ~np.isnan(routed['down_ref_seg']), other=routed['up_ref_seg'])).ffill('seg')
-    elif fill_method == 'r2':
-        pass
-    elif fill_method == 'leave_null':
-        pass
+    if fill_method == 'leave_null':
+        # since there should be no -1 segs from mizuroute, we can set nan's to -1 to acknowledge
+        # that they have been addressed and still set them apart from the rest of the data
+        routed['up_ref_seg'] = (routed['up_ref_seg'].where(~np.isnan(routed['up_ref_seg']), other=-1))
+        routed['down_ref_seg'] = (routed['down_ref_seg'].where(~np.isnan(routed['down_ref_seg']), other=-1))
     else:
-        raise ValueError('Invalid method provided for "fill_method"')
-
+        if fill_method == 'forward_fill':
```

Comment:
Shouldn't these just be `elif`s

Reply:
Ah, I did this so that in lines 209-213 I can check if there are nans that arise in a fill method that is not "leave_null" (where nans would be fine to be leftover). I can remove this check if you'd like, but it may get harder to trace where things went wrong if it is simply due to the fill_method

Reply:
Hmm, I see. I think that's the kind of thing that somebody might change in the future without understanding this. Instead, does it make sense to put these into `elif`s, but also change the check from 209-213 to have a check if `leave_null` was the fill method? Then if it was `leave_null` we just don't do that check?

Reply:
oh yeah, that'd work